src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -0,0 +1,493 @@
+use std::{iter, num::NonZeroUsize};
+
+use ff::{Field, FromUniformBytes, PrimeField, PrimeFieldBits};
+use halo2_proofs::circuit::Value;
+use halo2curves::{Coordinates, CurveAffine};
+use num_traits::Num;
+
+use crate::{
+    constants::NUM_CHALLENGE_BITS,
+    gadgets::{
+        ecc::{AssignedPoint, EccChip},
+        nonnative::bn::{
+            big_uint::{self, BigUint},
+            big_uint_mul_mod_chip::{self, BigUintMulModChip, ModOperationResult},
+        },
+    },
+    main_gate::{AssignedBit, AssignedValue, MainGate, MainGateConfig, RegionCtx, WrapValue},
+    nifs::CrossTermCommits,
+    plonk::{PlonkInstance, RelaxedPlonkInstance},
+    poseidon::ROCircuitTrait,
+    util,
+};
+
+pub(crate) struct FoldRelaxedPlonkInstanceChip<C: CurveAffine>
+where
+    C::Base: PrimeFieldBits + FromUniformBytes<64>,
+{
+    W: C,
+    E: C,
+    u: C::ScalarExt,
```

Comment:
with the separation of the challenge from polynomial query, the RelaxedPlonkInstance and PlonkInstance will be different than Nova:
e.g. here it will be:
{W:C, E:C, challenges: [C::ScalarExt; num_challenges], X0, X1}

Reply:
Notice that after I integrate the lookup arguments, there will be more challenges from lookup. So that the RelaxedPlonkInstance actually includes custom gate + lookup. We can treat them together.